lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -131,184 +134,208 @@ export default class Recipe extends RecipeModule {
     }
 
     // this function returns the user ID for which the session will be created.
-    createPrimaryUserIdOrLinkAccounts = async ({
+    createPrimaryUserIdOrLinkByAccountInfoOrLinkToSessionIfProvided = async ({
```

Comment:
in this function, why can't we just call the authCheck function and then based on if isFirstFactor = true, we call `tryLinkingByAccountInfo`, else we try and link to the session?